lib/utils/report.mjs

```diff
@@ -94,13 +97,21 @@ export default function report(problem) {
 	if (start) {
 		warningProperties.start = start;
 	} else if (isNumber(index)) {
-		warningProperties.index = index;
+		if (range?.start) {
+			warningProperties.start = range.start;
+		} else {
```

Comment:
`range.start` will now be more accurate for our use case than passing `index`.

We should use it when available.

Given that we calculate it from `index` it is also gated behind the check that `index` is a number.

Reply:
It is impossible to have an `index` value that is accurate and not to have a `range.start`.

A rule could create a report with an `index` and `line` while omitting the `node`, but then the `index` can't be used anyway.